apps/desktop/src/platform/services/renderer-ssh-agent.service.ts

```diff
@@ -50,66 +67,86 @@ export class SSHAgentService {
-              AuthenticationStatus.Locked
-            ) {
-              await new Promise((resolve) => setTimeout(resolve, 100));
-              if (new Date().getTime() - start.getTime() > 1000 * 60 * 5) {
-                this.logService.error("[SSH Agent] Timeout waiting for unlock");
-                this.toastService.showToast({
-                  variant: "error",
-                  title: null,
-                  message: this.i18nService.t("sshAgentUnlockTimeout"),
-                });
-                await ipc.platform.sshagent.signRequestResponse(id, false);
-                return;
-              }
+            const unlocked = firstValueFrom(
+              this.authService
+                .authStatusFor$(activeAccountId)
+                .pipe(filter((status) => status === AuthenticationStatus.Unlocked)),
+            );
+            const timeout = new Promise((_, reject) =>
+              setTimeout(reject, this.SSH_VAULT_UNLOCK_REQUEST_TIMEOUT),
+            );
+
+            try {
+              await Promise.race([unlocked, timeout]);
+            } catch (error) {
+              this.logService.error(error);
+              this.logService.error("[Ssh Agent] Timeout waiting for unlock");
+              this.toastService.showToast({
+                variant: "error",
+                title: null,
+                message: this.i18nService.t("sshAgentUnlockTimeout"),
+              });
+              await ipc.platform.sshAgent.signRequestResponse(messageId, false);
+              return;
             }
           }
 
           const decryptedCiphers = await this.cipherService.getAllDecrypted();
-          const cipher = decryptedCiphers.find((cipher) => cipher.id == cipherUuid);
+          const cipher = decryptedCiphers.find((cipher) => cipher.id == cipherId);
 
-          const dialogRef = ApproveSSHRequestComponent.open(
+          const dialogRef = ApproveSshRequestComponent.open(
             this.dialogService,
             cipher.name,
             this.i18nService.t("unknownApplication"),
           );
           const result = await firstValueFrom(dialogRef.closed);
-          await ipc.platform.sshagent.signRequestResponse(id, result);
+          await ipc.platform.sshAgent.signRequestResponse(messageId, result);
           ipc.platform.hideWindow();
-        })()
-          .then(() => {})
-          .catch((e) => {
-            this.logService.error("Error in SSHAgent sign request: ", e);
+        }),
+      )
+      .subscribe();
+
+    combineLatest([
+      timer(0, this.SSH_REFRESH_INTERVAL),
+      this.desktopSettingsService.sshAgentEnabled$,
+    ])
+      .pipe(
+        filter(([_, sshAgentEnabled]) => sshAgentEnabled),
+        concatMap(async () => {
+          const ciphers = await this.cipherService.getAllDecrypted();
+          if (ciphers == null) {
+            await ipc.platform.sshAgent.lock();
+            return;
+          }
```

Comment:
issue: sorry, I forgot about the path that was clearing keys!

```suggestion
    combineLatest([
      timer(0, this.SSH_REFRESH_INTERVAL),
      this.desktopSettingsService.sshAgentEnabled$,
    ])
      .pipe(
        concatMap(async ([_, sshAgentEnabled]) => {
          if (!sshAgentEnabled) {
            await ipc.platform.sshagent.setKeys([]);
            return;
          }

          const ciphers = await this.cipherService.getAllDecrypted();
          if (ciphers == null) {
            await ipc.platform.sshAgent.lock();
            return;
          }
```